.github/workflows/run_tests.yml

```diff
@@ -1,38 +1,45 @@
 name: Run Tests
 on:
-  pull_request:
-    branches:
-      - master
+    pull_request:
+        branches:
+            - master
 jobs:
-  build-and-test:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Install Go
-        uses: actions/setup-go@v2
-        with:
-          go-version: 1.21.x
+    build-and-test:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Install Go
+              uses: actions/setup-go@v2
+              with:
+                  go-version: 1.21.x
 
-      - name: Install Protoc
-        uses: arduino/setup-protoc@v1
-        with:
-          version: 3.x
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
+            - name: Install Protoc
+              uses: arduino/setup-protoc@v1
+              with:
+                  version: 3.x
+                  repo-token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Install Protoc Go plugin
-        run: go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
+            - name: Install Protoc Go plugin
+              run: go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
 
-      - name: Install Node
-        uses: actions/setup-node@v3
-        with:
-          node-version: '19'
+            - name: Install Node
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 'lts/*'
```

Comment:
This should be changed to 20, not LTS since everything locally is now documented as 20

Reply:
We can explicitly set it to 20 however my reasoning behind picking the latest LTS is:

A - run into the same thing that this replaces which is changing the engine version in package json but still keeping the previous (or previous previous) version in CICD because it hasn't needed to be looked at for a while.

B - If there is something that doesn't run on the latest stable LTS, that probably is a problem which should be addressed. 

This syntax does resolve the version currently to 20.4.0 


Thoughts?

Reply:
In rare occasions a mismatch in Node versions could introduce code that doesn't work anymore. It's rare, but happens.
So I prefer to set it to `20.15.0` which is the current 20.x LTS, and then make sure we check more precise when upgrading node versions.

Reply:
Changed to 20 now for `20.x`. Im working out a way to future proof the engines and node mismatch for my cicd pipelines so I'll probably end up making a quick github action one time to check it is compliant with the engines field or it will fail. It will be useful for projects that need to use npm (like electron projects). I tried to look at using `engine-strict` however it only works for upstream packages and not the root package.json you're using it on. It's a bit of a pain as pnpm has this functionality, and I think it makes sense to not implement it in npm but 🤷